bears/php/PHPCodeSnifferBear.py

```diff
@@ -0,0 +1,178 @@
+            lowercase.
+        :param blank_line_after_namespace_declaration:
+            Ensures that there is a blank line after a namespace declaration.
+        :param check_use_blocks:
+            Ensures that there is one blank line after a `use` block,
+            that there is only one use block per line, and that all `use`
+            declaration are done after namespaces declaration.
+        :param check_class_declaration:
+            Ensures that ``extends`` and ``implements`` keywords are declared
+            on the same line as the class name, that the opening brace for a
+            class is on the next line, and that the closing brace for a class
+            is on the next line after the body. Allows splitting implements
+            list accross multiple lines.
+        :param check_property_declaration:
+            Ensures that visibility is declared on all property, that the var
+            keyword is not used to declare a property, that there is not more
+            that one property declared on a line, that property are not
+            prefixed with an underscore.
+        :param force_scope_modifier_on_method:
+            Verifies that class methods have scope modifiers.
+        :param function_declaration_argument_spacing:
+            Number of spaces between arguments in function declaration.
+        :param allow_multiline_function_declaration:
+            Allows argument lists to be split accross multiple lines (correctly
+            indented.
+        '''
+        allow_tabs = ('<rule ref="Generic.WhiteSpace.DisallowTabIndent"/>'
+                      if use_spaces else '')
+        statements_per_line = (
+            '<rule ref="Generic.Formatting.DisallowMultipleStatements"/>'
+            if not allow_multiple_statements_per_line else '')
+        keywords_case = (
+            '<rule ref="Generic.PHP.LowerCaseKeyword"/>'
+            if force_lower_case_keywords else '')
+        constants_case = (
+            '<rule ref="Generic.PHP.LowerCaseConstant"/>'
+            if force_lower_case_constants else '')
+        namespace_declaration = (
+            'rule rel="PSR2.Namespaces.NamespaceDeclarationSniff"/>'
+            if blank_line_after_namespace_declaration else '')
+        use_declaration = (
+            'rule rel="PSR2.Namespaces.UseDeclarationSniff"/>'
+            if check_use_blocks else '')
+        class_declaration = (
+            'rule rel="PSR2.Classes.ClassDeclarationSniff"/>'
+            if check_class_declaration else '')
+        property_declaration = (
+            'rule rel="PSR2.Classes.PropertyDeclarationSniff"/>'
+            if check_property_declaration else '')
+        method_scope = (
+            '<rule ref="Squiz.Scope.MethodScope"/>'
+            if force_scope_modifier_on_method else '')
+        multiline_function_declaration = (
+            '<rule ref="Squiz.Functions.MultiLineFunctionDeclaration"/>'
```

Comment:
you should make out of that a format-string/function which you use inline in the "grand" formatting:
```python
def _rule(name):
    if name:
        rule_formatstring = '<rule ref="{}"/>'
        return rule_formatstring.format(name)
    else:
        return ''

configs = '''
...
'''
configs.format(use_spaces=_rule('Squiz.Functions.Spaces'), ...)
```
or you can beautify the template a bit, by not inserting a field for each rule, but a general 'rules' field:
```python
rule_map = {'Squiz.use_spaces': use_spaces}
rules = ['<rule ref="{}"/>'.format(k) for k, v in rule_map.items() if v]

configs = '''blablabla....
</rule>
{rules}
...
'''
configs.format(rules=''.join(rules))
```
many possibilities here :)